artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/buffer/TimedBuffer.java

```diff
@@ -288,7 +266,14 @@ private void flush(final boolean force) {
             buffer.getBytes(0, bufferToFlush);
 
             final List<IOCallback> ioCallbacks = callbacks == null ? Collections.emptyList() : callbacks;
-            bufferObserver.flushBuffer(bufferToFlush, pendingSyncs.get() > 0, ioCallbacks);
+
+            final boolean requiredSync = pendingSyncs.get() > 0;
+            this.flushProfiler.onStartFlush(bufferToFlush.position(), requiredSync);
```

Comment:
why not a guard statement here? so if disabled the JVM would simply compile out the code when disabled.